docpages/example_programs/the_basics/using_callback_functions.md

```diff
@@ -0,0 +1,97 @@
+                    return;
+                }
+
+                // callback.value will contain the message we are getting but the type will be std::variant, so we will have to get the message itself.
+                auto message = callback.get <dpp::message>();
+                //std::get <dpp::message>(callback.value) would give the same result
+
+                event.reply(message); // replies with the same message
+            });
+        }
+        if (event.command.get_command_name() == "channel-create") {
+
+            //create a text channel
+            dpp::channel channel = dpp::channel()
+                .set_name("test")
+                .set_type(dpp::channel_type::CHANNEL_TEXT)
+                .set_guild_id(1112789761032200306)
+                .set_parent_id(0);
+            bot.channel_create(channel, [&bot, event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) { // catching an error to log it
+                    bot.log(dpp::loglevel::ll_error, callback.get_error().message);
+                }
+
+                // callback.value will contain the channel we are creating but the type will be std::variant, so we will have to get the channel itself.
+                auto channel = callback.get <dpp::channel>();
+                // std::get <dpp::channel>(callback.value) would give the same result
+
+                // reply with the created channel information
+                dpp::message message = dpp::message(fmt::format("The channel's name is {0}, ID is {1} and type is {2}", channel.name, channel.id, channel.get_type()));
+                /* note that channel types are represented as numbers */
+                event.reply(message);
+            });
+        }
+        if (event.command.get_command_name() == "msg-error") {
+            bot.message_get(69, 420, [event](const dpp::confirmation_callback_t& callback) -> void {
+
+                // the error will occur since there is no message with ID '69' that is in a channel with ID '420' (at least the bot can't see it)
+                if (callback.is_error()) {
+                    event.reply(callback.get_error().message);
+                    return;
+                }
+
+                // we won't be able to get here because of the return; statement
+                auto message = callback.get <dpp::message>();
+                event.reply(message);
+            });
+        }
+    });
+
+    bot.on_ready([&bot](const dpp::ready_t& event) {
+        if (dpp::run_once <struct register_global_commands>()) {
+            dpp::slashcommand msg_get("msg-get", "Get the message", bot.me.id);
+            dpp::slashcommand channel_create("channel-create", "Create a channel", bot.me.id);
+            dpp::slashcommand msg_error("msg-error", "Get an error instead of message :)", bot.me.id);
+
+            bot.global_bulk_command_create({msg_get, channel_create, msg_error});
```

Comment:
This breaks the [code convention for lists](https://dpp.dev/coding-standards.html#:~:text=braces%20with%20statements!-,Lists,-Lists%20should%20have). Look at [this page](https://dpp.dev/slashcommands.html#:~:text=bot.global_bulk_command_create(%7B%20pingcommand%2C%20pongcommand%2C%20dingcommand%2C%20dongcommand%20%7D)%3B) if you need help understanding.

Reply:
My bad, I'll look

Reply:
Ez

Reply:
*Presses space key twice*